### PR TITLE
logging: non-blocking does not enqueue new messages when buffer is full

### PR DESCRIPTION
**Pull request by PettitWesley**
Fixes https://github.com/moby/moby/issues/45200

### CODE REVIEW

File: config/containers/logging/configure.md [hunk 1]
```diff
@@ -135,24 +135,22 @@ Docker provides two modes for delivering messages from the container to the log
 driver:
 
 * (default) direct, blocking delivery from container to driver
-* non-blocking delivery that stores log messages in an intermediate per-container
-  ring buffer for consumption by driver
+* non-blocking delivery that stores log messages in an intermediate per-container buffer for consumption by driver
```
Comment (PettitWesley) [inline config/containers/logging/configure.md:138]: I'm guessing this edit will be contentious and I can remove it if you want. But with the way the code actually works it is not a true ring buffer, its just a buffer: https://github.com/moby/moby/blob/master/daemon/logger/ring.go#L160

**Review by cpuguy83 — APPROVED** [review]

File: config/containers/logging/configure.md (hunk 1 above)
Comment (cpuguy83) [reply config/containers/logging/configure.md:138]: Nah, this is perfect. It describes reality. I don't recall how these docs got written but they are definitely wrong.

**Review by thaJeztah — APPROVED** [review]
LGTM, thank you!

**Comment by thaJeztah** [comment]
And, it's merged! Changes won't be visible yet on docs.docker.com, but will be part of the next deploy (probably on Monday)

**Pull request merged**